### PR TITLE
WT-8439 - test_wt8246_compact_rts_data_correctness failed with invalid argument

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
This test expects that the compact process is killed before completion. For some reason in OSX machines the compact process completes, so reducing the TIMEOUT time before killing compact would help.

### CODE REVIEW

**Comment by sueloverso** [comment]
I have a bunch of questions.
Were you able to reproduce this on MacOS to confirm that it was completing unexpectedly?
If so, are you sure that 2 seconds is the right value to choose for MacOS?
Should that define setting to 2 be specific to MacOS? I'm concerned that a smaller value might be too short for PPC or ZSeries hardware that is very slow in that the work may not have gotten started yet on those machines.

Timing defines are often hard. Is there any other way such as using sentinel files to indicate the test has started or is about to finish?

**Comment by raviprakashgiri29** [comment]
> I have a bunch of questions. Were you able to reproduce this on MacOS to confirm that it was completing unexpectedly? If so, are you sure that 2 seconds is the right value to choose for MacOS? Should that define setting to 2 be specific to MacOS? I'm concerned that a smaller value might be too short for PPC or ZSeries hardware that is very slow in that the work may not have gotten started yet on those machines.
> 
> Timing defines are often hard. Is there any other way such as using sentinel files to indicate the test has started or is about to finish?

@sueloverso Thanks for the comments. I have implemented a solution where I create a sample file in `session_compact.c` (in diagnostic mode) just before starting compact operation and then check for the file in the test parent process to know that the compact process has started and then the parent process can kill the child process. If this solution does not sound right then I have another solution in my mind to keep the older code of creating the `checkpoint_started` file in the test and introduce timing stress in the compact.

File: src/session/session_compact.c [hunk 1]
```diff
@@ -245,6 +248,16 @@ __compact_worker(WT_SESSION_IMPL *session)
      */
     WT_ERR(__compact_checkpoint(session));
 
+#ifdef HAVE_DIAGNOSTIC
+    /*
+     * Create the compact_started file so that the test can start its timer.
+     */
+    WT_ERR(
+      __wt_snprintf(compact_file, sizeof(compact_file), "%s/compact_started", S2C(session)->home));
```
Comment (sueloverso) [inline src/session/session_compact.c:256]: We cannot do this. WT does not own that namespace. The user may have a file of the same name.

**Comment by raviprakashgiri29** [comment]
@sueloverso  I added the compact timing stress in this test so once the child process starts the compact then it can wait only for 1 second and then kill the compact and do the data validation. I tested this change in almost all machines by enabling compact verbose messages just to confirm that we kill the compact process when it is in progress.

File: test/csuite/wt8246_compact_rts_data_correctness/main.c [hunk 2]
```diff
@@ -255,6 +255,8 @@ workload_compact(const char *home, const char *table_config, const char *uri)
 
     testutil_check(session->checkpoint(session, NULL));
 
+    testutil_check(session->checkpoint(session, "force"));
```
Comment (sueloverso) [inline test/csuite/wt8246_compact_rts_data_correctness/main.c:258]: Why do we need to force a checkpoint immediately after doing a checkpoint? If this is needed please add a comment explaining why we want the forced checkpoint.

**Review by sueloverso — APPROVED** [review]
I have one request for a comment. This has not passed PR testing due to system issues.

File: test/csuite/wt8246_compact_rts_data_correctness/main.c (hunk 2 above)
Comment (raviprakashgiri29) [reply test/csuite/wt8246_compact_rts_data_correctness/main.c]: Added a comment, thanks

**Pull request merged**